addons/payment_sips/models/sips.py

```diff
@@ -0,0 +1,237 @@
+            # Test key provided by Atos Wordline works only with version 1
+            merchant_id = '002001000000001'
+            key_version = '1'
+
+        sips_tx_values = dict(tx_values)
+        sips_tx_values.update({
+            'Data': u'amount=%s|' % amount +
+                    u'currencyCode=%s|' % currency_code +
+                    u'merchantId=%s|' % merchant_id +
+                    u'normalReturnUrl=%s|' % urlparse.urljoin(base_url, SipsController._return_url) +
+                    u'transactionReference=%s|' % tx_values['reference'] +
+                    u'keyVersion=%s' % key_version,
+            'InterfaceVersion': 'HP_2.3',
+        })
+
+        return_context = {}
+        if sips_tx_values.get('return_url'):
+            return_context[u'return_url'] = u'%s' % sips_tx_values.pop('return_url')
+        return_context[u'reference'] = u'%s' % sips_tx_values['reference']
+        sips_tx_values['Data'] += u'|returnContext=%s' % (json.dumps(return_context))
+
+        shasign = self._sips_generate_shasign(acquirer, sips_tx_values)
+        sips_tx_values['Seal'] = shasign
+        return partner_values, sips_tx_values
+
+    def sips_get_form_action_url(self):
+        return self._get_sips_urls(self.environment)['sips_form_url']
+
+
+class TxSips(models.Model):
+    _inherit = 'payment.transaction'
+
+    # sips status
+    _sips_valid_tx_status = ['00']
+    _sips_wait_tx_status = ['90', '99']
+    _sips_refused_tx_status = ['05', '14', '34', '54', '75', '97']
+    _sips_error_tx_status = ['03', '12', '24', '25', '30', '40', '51', '63', '94']
+    _sips_pending_tx_status = ['60']
+    _sips_cancel_tx_status = ['17']
+
+    # --------------------------------------------------
+    # FORM RELATED METHODS
+    # --------------------------------------------------
+    def _sips_data_to_object(self, data):
+        res = {}
+        for element in data.split('|'):
+            element_split = element.split('=')
+            res[element_split[0]] = element_split[1]
+        return res
+
+    @api.model
+    def _sips_form_get_tx_from_data(self, data):
+        """ Given a data dict coming from sips, verify it and find the related
+        transaction record. """
+
+        data = self._sips_data_to_object(data.get('Data'))
+        reference = data.get('transactionReference')
+
+        if not reference:
+            custom = json.loads(data.pop('returnContext', False) or '{}')
+            reference = custom.get('reference')
+
+        tx_ids = self.env['payment.transaction'].search([
+            ('reference', '=', reference), ])
+        if not tx_ids or len(tx_ids) > 1:
+            error_msg = 'Sips: received data for reference %s' % reference
+            if not tx_ids:
+                error_msg += '; no order found'
+            else:
+                error_msg += '; multiple order found'
+            _logger.error(error_msg)
+            raise ValidationError(error_msg)
+        return self.browse(tx_ids[0])
```

Comment:
In new API, you search returned a recordset, so no need to rebrowse it, so you can simply ``return tx_ids``